insights/parsers/tests/test_initscript.py

```diff
@@ -0,0 +1,152 @@
+#
+# chkconfig: 345 97 03
+# description:  This is a daemon which handles the task of connecting \
+#               periodically to the Spacewalk servers to \
+#               check for updates, notifications and perform system \
+#               monitoring tasks according to the service level that \
+#               this server is subscribed for
+#
+# processname: rhnsd
+# pidfile: /var/run/rhnsd.pid
+#
+
+echo script_chkconfig
+""".strip()
+
+LSB_SCRIPT = "etc/rc.d/init.d/script_lsb"
+LSB_CONTENT = """
+#!/bin/sh
+### BEGIN INIT INFO
+# Provides: rhnsd
+# Required-Start: $local_fs $network $remote_fs $named $time
+# Required-Stop: $local_fs $network $remote_fs $named
+# Default-Start: 2 3 4 5
+# Default-Stop: 0 1 6
+# Short-Description: Starts the Spacewalk Daemon
+# Description: This is a daemon which handles the task of connecting
+#               periodically to the Spacewalk servers to
+#               check for updates, notifications and perform system
+#               monitoring tasks according to the service level that
+#               this server is subscribed for.
+### END INIT INFO
+
+echo script_lsb
+""".strip()
+
+CHKCONFIG_LSB_SCRIPT = "etc/rc.d/init.d/script_chkconfig+lsb"
+CHKCONFIG_LSB_CONTENT = """
+#!/bin/sh
+#
+# rhnsd:        Starts the Spacewalk Daemon
+#
+# chkconfig: 345 97 03
+# description:  This is a daemon which handles the task of connecting \
+#               periodically to the Spacewalk servers to \
+#               check for updates, notifications and perform system \
+#               monitoring tasks according to the service level that \
+#               this server is subscribed for
+#
+# processname: rhnsd
+# pidfile: /var/run/rhnsd.pid
+#
+
+### BEGIN INIT INFO
+# Provides: rhnsd
+# Required-Start: $local_fs $network $remote_fs $named $time
+# Required-Stop: $local_fs $network $remote_fs $named
+# Default-Start: 2 3 4 5
+# Default-Stop: 0 1 6
+# Short-Description: Starts the Spacewalk Daemon
+# Description: This is a daemon which handles the task of connecting
+#               periodically to the Spacewalk servers to
+#               check for updates, notifications and perform system
+#               monitoring tasks according to the service level that
+#               this server is subscribed for.
+### END INIT INFO
+
+echo script_chkconfig+lsb
+""".strip()
+
+EMPTY_SCRIPT = "etc/rc.d/init.d/script_empty"
+EMPTY_CONTENT = """
+""".strip()
+
+HINTSONLY_SCRIPT = "etc/rc.d/init.d/script_hintsonly"
+HINTSONLY_CONTENT = """
+#!/bin/sh
+#
+
+case "$1" in
+start)
+        ;;
+stop)
+        ;;
+esac
+""".strip()
+
+
+class TestInitScript(unittest.TestCase):
+    def test_initscript1(self):
+        context = context_wrap(NOTINITSCRIPT_CONTENT)
+        context.path = NOTINITSCRIPT_SCRIPT
+
+        with pytest.raises(NotInitscriptException) as e_info:
+            r = InitScript(context)
+            self.assertIsNone(r)
+            assert context.path in str(e_info.value)
```

Comment:
Once the exception is thrown on line 113, lines 114 and 115 will not execute.  If the exception is not thrown then an assertion will be raised.  You could probably just remove line 114 as it is implied by the exception being raised, and move line 115 out one level to be checked after the exception is thrown.